src/Transactions/TransferWithMemo.cs

```diff
@@ -18,6 +20,36 @@ public sealed record TransferWithMemo(CcdAmount Amount, AccountAddress Receiver,
     /// </summary>
     private const byte TransactionType = (byte)Types.TransactionType.TransferWithMemo;
 
+    /// <summary>
+    /// Prepares the account transaction payload for signing.
+    /// </summary>
+    /// <param name="sender">Address of the sender of the transaction.</param>
+    /// <param name="sequenceNumber">Account sequence number to use for the transaction.</param>
+    /// <param name="expiry">Expiration time of the transaction.</param>
+    public PreparedAccountTransaction Prepare(
+        AccountAddress sender,
+        AccountSequenceNumber sequenceNumber,
+        Expiry expiry
+    ) => new(sender, sequenceNumber, expiry, this._transactionCost, this);
+
+    /// <summary>
+    /// The transaction specific cost for submitting this type of
+    /// transaction to the chain.
+    ///
+    /// This should reflect the transaction-specific costs defined here:
+    /// https://github.com/Concordium/concordium-base/blob/78f557b8b8c94773a25e4f86a1a92bc323ea2e3d/haskell-src/Concordium/Cost.hs
+    /// </summary>
+    private readonly EnergyAmount _transactionCost = new(300);
+
+    /// <summary>
+    /// Gets the size (number of bytes) of the payload.
+    /// </summary>
+    internal override PayloadSize Size() => new(
+        this.Memo.Length() +
```

Comment:
Here `this.Memo.Length()` is used and below `OnChainData.MaxLength`.

Is that correct?

Reply:
I removed the `Serialize` methods since they were private and kind of clunky. Now the `MemoryStream` just uses `Size().Size` instead, so it's defined in a single place.

Reply:
I removed the `Serialize` methods since they were private anyways. This lets us define the MemoryStream directly from the `Size` and solves this inconsistency.